packages/belos/tpetra/test/BlockGmres/test_bl_gmres_hb.cpp

```diff
@@ -0,0 +1,247 @@
+#include <Kokkos_DefaultNode.hpp>
+
+using namespace Teuchos;
+using Tpetra::Operator;
+using Tpetra::CrsMatrix;
+using Tpetra::MultiVector;
+using std::endl;
+using std::cout;
+using std::vector;
+using Teuchos::tuple;
+
+int main(int argc, char *argv[]) {
+
+  typedef double                           ST;
+  typedef ScalarTraits<ST>                SCT;
+  typedef SCT::magnitudeType               MT;
+  typedef Tpetra::Operator<ST,int>         OP;
+  typedef Tpetra::MultiVector<ST,int>      MV;
+  typedef Belos::OperatorTraits<ST,MV,OP> OPT;
+  typedef Belos::MultiVecTraits<ST,MV>    MVT;
+
+  GlobalMPISession mpisess(&argc,&argv,&cout);
+
+  bool success = false;
+  bool verbose = false;
+  try {
+    const ST one  = SCT::one();
+
+    int MyPID = 0;
+
+    typedef Tpetra::DefaultPlatform::DefaultPlatformType           Platform;
+    typedef Tpetra::DefaultPlatform::DefaultPlatformType::NodeType Node;
+
+    Platform &platform = Tpetra::DefaultPlatform::getDefaultPlatform();
+    RCP<const Comm<int> > comm = platform.getComm();
+    RCP<Node>             node = platform.getNode();
+
+    //
+    // Get test parameters from command-line processor
+    //
+    bool proc_verbose = false;
+    bool debug = false;
+    int frequency = -1;  // how often residuals are printed by solver
+    int numrhs = 1;      // total number of right-hand sides to solve for
+    int blocksize = 1;   // blocksize used by solver
+    int maxiters = -1;   // maximum number of iterations for solver to use
+    std::string filename("bcsstk14.hb");
+    MT tol = 1.0e-5;     // relative residual tolerance
+
+    CommandLineProcessor cmdp(false,true);
+    cmdp.setOption("verbose","quiet",&verbose,"Print messages and results.");
+    cmdp.setOption("debug","nodebug",&debug,"Run debugging checks.");
+    cmdp.setOption("frequency",&frequency,"Solvers frequency for printing residuals (#iters).");
+    cmdp.setOption("tol",&tol,"Relative residual tolerance used by Gmres solver.");
+    cmdp.setOption("filename",&filename,"Filename for Harwell-Boeing test matrix.");
+    cmdp.setOption("num-rhs",&numrhs,"Number of right-hand sides to be solved for.");
+    cmdp.setOption("max-iters",&maxiters,"Maximum number of iterations per linear system (-1 := adapted to problem/block size).");
+    cmdp.setOption("block-size",&blocksize,"Block size to be used by the Gmres solver.");
+    if (cmdp.parse(argc,argv) != CommandLineProcessor::PARSE_SUCCESSFUL) {
+      return -1;
+    }
+    if (debug) {
+      verbose = true;
+    }
+    if (!verbose) {
+      frequency = -1;  // reset frequency if test is not verbose
+    }
+
+    MyPID = rank(*comm);
+    proc_verbose = ( verbose && (MyPID==0) );
+
+    if (proc_verbose) {
+      std::cout << Belos::Belos_Version() << std::endl << std::endl;
+    }
+
+    //
+    // Get the data from the HB file and build the Map,Matrix
+    //
+    RCP<CrsMatrix<ST,int> > A;
+    Tpetra::Utils::readHBMatrix(filename,comm,node,A);
```

Comment:
A while back, I fixed this interface so that you could pass in `Teuchos::null` for the `node`.  That would mean you wouldn't need to create a Node instance (see above).